File: go/pangram/pangram.go
package pangram

import (
	"regexp"
	"strings"
)

func isLetter(s string) bool {
	isAlpha := regexp.MustCompile(`[A-Za-z]`).MatchString
	return isAlpha(s)
}

func IsPangram(s string) bool {
	var letters = map[string]bool{}
	chars := strings.Split(s, "")
	for _, i := range chars {
		if isLetter(i) {
			i = strings.ToLower(i)
			letters[i] = true
		}
	}
	if len(letters) == 26 {
		return true
	}
	return false
}

File: ruby/pangram/pangram.rb
class Pangram

  def self.pangram?( string )
    string = string.gsub(/[^A-Za-z]/, '')
    string.downcase!
    return false if string.length < 26
    letters = {}
    string.chars.each do |letter|
      letters[letter] = true
    end
    return letters.length == 26 ? true : false
  end


end

module BookKeeping
  VERSION = 6
end
